config/v1/types_dns.go

```diff
@@ -74,6 +80,16 @@ type DNSZone struct {
 	Tags map[string]string `json:"tags,omitempty"`
 }
 
+// Upstream defines the schema for associating a base domain with a list of name resolvers.
+type Upstream struct {
+	// domain is the base domain to match to forward name queries.
+	Domain string `json:"domain"`
+	// resolverList is a list of name servers responsible for resolving name queries from Domain.
+	// Each resolver in the list is represented by an IP address or IP:port if the resolver
+	// listens on a UDP port other than 53.
+	ResolverList []string `json:"resolverList"`
```

Comment:
Is there any standardized validation around address references we can refer to or codify through OpenAPI validations?

Will this be compatible with ipv6 addresses?

Reply:
While IPv4/IPv6 formats are not defined by the [OpenAPI Specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#dataTypes), they are defined by the [JSON Schema](https://tools.ietf.org/html/draft-wright-json-schema-validation-00#section-7.3) which is what OpenAPI's Schema Object is closely based on.